test/Microsoft.Azure.WebJobs.Extensions.Kafka.UnitTests/DiagnosticTest.cs

```diff
@@ -0,0 +1,117 @@
+            Assert.Equal(consumerGroup, activity.GetTagItem(ActivityTags.KafkaConsumerGroup).ToString());
+            Assert.Equal(kafkaEvent.Partition.ToString(), activity.GetTagItem(ActivityTags.KafkaPartition));
+            var key = activity.GetTagItem(ActivityTags.KafkaMessageKey).ToString();
+            Assert.Equal(kafkaEvent.Key, activity.GetTagItem(ActivityTags.KafkaMessageKey).ToString());
+            singleEventActivityProvider.Dispose();
+            activityListener.Dispose();
+        }
+
+        [Fact]
+        public void BatchActivityProvider_Should_Create_Activity_With_Links()
+        {
+            var consumerGroup = "my-consumer-group";
+            int numkafkaEvents = 10;
+            var kafkaEvents = new KafkaEventData<string, string>[numkafkaEvents];
+            var topicName = "mytopic";
+            var numActivityTags = 5;
+            for (int i = 0; i< numkafkaEvents; i++)
+            {
+                kafkaEvents[i] = CreateKafkaEventObjWithTraceParentHeader(topicName);
+            }
+
+            var activityListener = new ActivityListener
+            {
+                ShouldListenTo = _ => true,
+                Sample = (ref ActivityCreationOptions<ActivityContext> _) => ActivitySamplingResult.AllData,
+            };
+            ActivitySource.AddActivityListener(activityListener);
+            var batchEventActivityProvider = new BatchEventActivityProvider(kafkaEvents, consumerGroup);
+            Assert.NotNull(batchEventActivityProvider.Activity);
+            var activity = batchEventActivityProvider.Activity;
+            var activityLinks = activity.Links.ToArray<ActivityLink>();
+
+            Assert.Equal(kafkaEvents.Length, activityLinks.Length);
+
+            for ( int i=0; i<kafkaEvents.Length; i++)
+            {
+                var kafkaEvent = kafkaEvents[i];
+                GetTraceIdAndSpanIdFromKafkaHeader(kafkaEvent, out var traceId, out var spanId);
+                Assert.Equal(traceId, activityLinks[i].Context.TraceId.ToString());
+                Assert.Equal(spanId, activityLinks[i].Context.SpanId.ToString());
+            }
+
+            Assert.Equal(numActivityTags, activity.Tags.Count());
+            Assert.Equal("kafka", activity.GetTagItem(ActivityTags.System));
+            Assert.Equal(topicName, activity.GetTagItem(ActivityTags.DestinationName));
+            Assert.Equal("topic", activity.GetTagItem(ActivityTags.DestinationKind));
+            Assert.Equal("process", activity.GetTagItem(ActivityTags.Operation));
+            Assert.Equal(consumerGroup, activity.GetTagItem(ActivityTags.KafkaConsumerGroup));
+            batchEventActivityProvider.Dispose();
+            activityListener.Dispose();
+        }
+
+        public string CreateRandomTraceParent()
+        {
+            var traceId = ActivityTraceId.CreateRandom();
+            var spanId = ActivitySpanId.CreateRandom();
+            string traceparent = "00-" + traceId + "-" + spanId + "-" + "01";
```

Comment:
nit pick : return should directly from this statement as traceparent is not being used anywhere

Reply:
Done